File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if nums[0] < nums[-1]:
            return nums[0]

        left, right = 0, len(nums) - 1

        while left <= right:
            middle = (left + right) // 2
            if nums[middle] < nums[middle - 1]:
                return nums[middle]
            elif nums[middle] > nums[right]:
                left = middle + 1
            else:
                right = middle - 1
            
        return nums[0]

File: Data Structures & Algorithms/is-palindrome/submission-1.cpp
class Solution {
private:
    string me(string s) {
        for (int i = 0; i < s.length(); i++) {
            if (!isalnum(s[i])) {
                s.erase(s.begin() + i);
                i--;
            }
            if (isupper(s[i])) s[i] = tolower(s[i]);
        }
        return s;
    }
public:
    bool isPalindrome(string s) {
        s = me(s);

        int left = 0;
        int right = s.length() - 1;

        cout<<s;
        while (left != right && left < right) {
            if (tolower(s[left]) != tolower(s[right])) {
                cout<<left;
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};
